### PR TITLE
WX-1153 Azure blob read md5 from metadata for large files

### PR DESCRIPTION
**Pull request by aednichols**

### CODE REVIEW

File: filesystems/blob/src/test/scala/cromwell/filesystems/blob/BlobPathBuilderSpec.scala [hunk 1]
```diff
@@ -100,6 +100,30 @@ class BlobPathBuilderSpec extends AnyFlatSpec with Matchers with MockSugar {
...
+  }
+
+  it should "choose the native md5 over the metadata md5 for files that have both" in {
+    val builder = makeBlobPathBuilder(endpoint, store)
+    val evalPath = "/test/redundant_md5_test.txt"
+    val testString = endpoint.value + "/" + store + evalPath
+    val blobPath1: BlobPath = builder build testString getOrElse fail()
+    blobPath1.md5HexString.toOption.get should equal(Some("9e5ceec07c8730b593a3a4b4ae324475"))
+  }
+
```
Comment (aednichols) [inline filesystems/blob/src/test/scala/cromwell/filesystems/blob/BlobPathBuilderSpec.scala:126]: I'm not really happy with immediately `ignore`ing these tests, gonna mess around to see if they can be made to work in CI.

File: filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilder.scala [hunk 2]
```diff
@@ -116,16 +128,32 @@ case class BlobPath private[blob](pathString: String, endpoint: EndpointURL, con
...
+    // is rendered in base64 in the Azure web portal.
+    def hexString(bytes: Array[Byte]): String = bytes.map("%02x".format(_)).mkString
+
+    blobFileAttributes.map { attr: AzureBlobFileAttributes =>
+      (Option(attr.blobHttpHeaders().getContentMd5), md5FromMetadata) match {
+        case (None, None) => None
+        case (None, Some(metadataMd5)) => Option(metadataMd5)
+        case (Some(headerMd5Bytes), None) if headerMd5Bytes.isEmpty => None
+        // (Some, Some) could happen if an uploader redundantly populates an md5 for a small file.
+        // Doesn't seem like an erroneous condition so just choose the native one.
```
[the remark this answers is not in the document]
Comment (aednichols) [reply filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilder.scala]: Implemented the decision from today's discussion.

File: filesystems/blob/src/test/scala/cromwell/filesystems/blob/BlobPathBuilderSpec.scala [hunk 3]
```diff
@@ -100,6 +100,30 @@ class BlobPathBuilderSpec extends AnyFlatSpec with Matchers with MockSugar {
...
   }
 
+  it should "read md5 from small files <5g" in {
+    val builder = makeBlobPathBuilder(endpoint, store)
+    val evalPath = "/test/inputFile.txt"
+    val testString = endpoint.value + "/" + store + evalPath
+    val blobPath1: BlobPath = builder build testString getOrElse fail()
+    blobPath1.md5HexString.toOption.get should equal(Some("967f0f086992f1a8b48f0a533f80290b"))
+  }
+
```
[the remark this answers is not in the document]
Comment (aednichols) [reply filesystems/blob/src/test/scala/cromwell/filesystems/blob/BlobPathBuilderSpec.scala:104]: Now that the tests are working... with pleasure!

**Review by THWiseman — APPROVED** [review]
LGTM. Love having these tests working!

**Pull request merged**